lib/one_wire/ibutton/protocols/dallas/protocol_ds1971.c

```diff
@@ -0,0 +1,258 @@
+#include "protocol_ds1971.h"
+
+#include <core/core_defines.h>
+#include <toolbox/pretty_format.h>
+
+#include "dallas_common.h"
+
+#define DS1971_FAMILY_CODE 0x14U
+#define DS1971_FAMILY_NAME "DS1971"
+
+#define DS1971_SRAM_DATA_SIZE 32U
+#define DS1971_SRAM_PAGE_SIZE 32U
+#define DS1971_COPY_SCRATCH_TIMEOUT_US 250U
+
+#define DS1971_DATA_BYTE_COUNT 4U
+
+#define DS1971_SRAM_DATA_KEY "Sram Data"
+
+#define DS1971_CMD_READ_MEMORY 0xF0
```

Comment:
Use `DALLAS_COMMON_CMD_` macros if the required command codes were already defined in `dallas_common.h`